[Leonardo Wolf] iP

leowolf275 opened the pull request:
### Boss

> Your special Boss that is going to organize your tasks for you! 🔥

Boss frees your mind of having to remember things you need to do. 

- text-based
- easy to learn
- SUPER _FAST_ to use

Here are the following steps to follow:

1. Clone the GitHub repo [here](https://github.com/leowolf275/ip)
2. Run the program
3. Add tasks and **enjoy**!

Features:

- [ ] Manage Todo Tasks
- [ ] Manage Deadlines
- [ ] Manage Events

Take a look at the main method!
``` 
public static void main(String[] args) {
   new Boss("src/main/data/boss.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

leowolf275 committed:
Level-0: Rename, Greet, Exit

leowolf275 committed:
Level-1: Echo

leowolf275 committed:
Level 2: Add, List

leowolf275 committed:
Level 3: Mark as Done

leowolf275 committed:
Level 4. ToDos, Events, Deadlines

leowolf275 committed:
Level-4 improvements

leowolf275 committed:
A-TextUiTesting

leowolf275 committed:
Level-5 and Level-6

leowolf275 committed:
A-Enums

leowolf275 committed:
Add Level-7: Save feature

leowolf275 committed:
Merge branch 'branch-Level-7'

leowolf275 committed:
Add Level-8: Dates and Times feature

leowolf275 committed:
Merge branch 'branch-level-8'

leowolf275 committed:
Add A-More-OOP: Use more OOP

leowolf275 committed:
Merge branch 'a-moreOOP'

leowolf275 committed:
Merge branch 'add-gradle-support'

leowolf275 committed:
Add A-JUnit Tests

leowolf275 committed:
Merge branch 'A-JUnit'

leowolf275 committed:
Add JavaDoc to code

leowolf275 committed:
Add Level-9: Find

leowolf275 committed:
Follow the Coding Standard

leowolf275 committed:
Add more JavaDoc

leowolf275 committed:
Merge branch 'A-JavaDoc'

leowolf275 committed:
Merge branch 'A-CodingStandard'

leowolf275 committed:
Merge branch 'Level-9'

leowolf275 committed:
Handle case where file does not exist

leowolf275 committed:
Add more OOP improvements

leowolf275 committed:
Add further OOP improvements

leowolf275 committed:
Remove two print statements

kimberlytmq commented on src/main/java/boss/Boss.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+
+    /**
```
Nice use of comments across all your code! It makes it very readable and easy to understand.

kimberlytmq commented on src/main/java/boss/TimeTask.java:
```diff
@@ -0,0 +1,76 @@
+/**
+ * Represents an event associated with times!
+ */
+public class TimeTask extends Task {
```
Is it necessary to create a TimeTask class? Or could you have Deadline and Event ineherit from Task? Could be an alternative way to do things that you can try!

SuhailLoya replied:
Perhaps the functions here can be implemented in parser

kimberlytmq commented on src/main/java/boss/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+
+        String newFileData = getAll(tasks);
+
+        System.out.println("Ok! I have marked this task as not done yet!");
```
Maybe you can consider putting these print statements in the UI class?

kimberlytmq commented on src/main/java/boss/Parser.java:
```diff
@@ -0,0 +1,129 @@
+            case TODO:
+                String[] string = task.split(" ");
+                if (string.length == 1) {
+                    ui.showLoadingError(new BossException("The description of todo cannot be empty!"));
```
An interesting way to implement different error messages! Could it be made neater by creating a few set error messages?

kimberlytmq commented on src/main/java/boss/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+    /**
+     * Prints user messages to the screen.
+     */
+    public void printabstraction() {
```
Can change to camelcasing!

SuhailLoya commented on src/main/java/boss/Boss.java:
```diff
@@ -0,0 +1,74 @@
+     *  will read and write data from and to respectively.
+     */
+
+    public Boss(String filePath) {
```
Consider removing empty line at line 52 to better adhere to the coding standard :) I noticed the same issue at several other places too

> No blank line between the documentation block and the method/class.

SuhailLoya commented on src/main/java/boss/Boss.java:
```diff
@@ -0,0 +1,74 @@
+
+    public static void main(String[] args) {
+        new Boss("src/main/data/boss.txt").run();
+    }
```
I like the flexibility of Boss class to be able to accept different paths to data file

SuhailLoya commented on src/main/java/boss/BossException.java:
```diff
@@ -0,0 +1,22 @@
+
+    /**
+     * Creates a Boss Exception object
+     * @param message contains error message of exception
```
Consider adding an empty line between the description and parameter section, and also add a punctuation behind each parameter description. I noticed the same issue at many other places too

> Empty line between description and parameter section.
> Punctuation behind each parameter description.

SuhailLoya commented on src/main/java/boss/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+        super(description, isDone);
+        this.by = by;
+
+        setDateTime(by);
```
I like the use of setter here. This way if there is a change to the way date time is set, you only need to change it at the setter :)

SuhailLoya commented on src/main/java/boss/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+/**
+ * Represents a Deadline, which is a type of task
+ * users can add to their list.
+ */
```
Perhaps can be a bit more descriptive by informing that deadline task has a by date or a due date. 
Same goes for other subclasses of Task

SuhailLoya commented on src/main/java/boss/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+ * users can add to their list.
+ */
+public class Deadline extends TimeTask {
+    protected String by;
```
should this be made private with getters and setters? Same goes for other protected variables

SuhailLoya commented on src/main/java/boss/Event.java:
```diff
@@ -0,0 +1,58 @@
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+            this.toTime = LocalDateTime.parse(to, formatter);
+            this.to = toTime.format(DateTimeFormatter.ofPattern("MMM d yyyy HH:mm "));
+        }
```
should this be extracted out to a function like in the case of deadline task?

SuhailLoya commented on src/main/java/boss/Parser.java:
```diff
@@ -0,0 +1,129 @@
+        } else if (task.startsWith("find")) {
+            String description = task.split("find ")[1];
+            tasks.find(description);
+        }
```
I like the way of parsing commands here. One suggestion is perhaps the command list can be extracted out as an enum or constants

SuhailLoya commented on src/main/java/boss/Task.java:
```diff
@@ -0,0 +1,67 @@
+ * Represents the Task class.
+ */
+
+public class Task {
```
I like the descriptive docs and function names

SuhailLoya commented on src/main/java/boss/Parser.java:
```diff
@@ -0,0 +1,129 @@
+                tasks.printabstraction();
+
+                storage.writeToFile(newItem + System.lineSeparator(), true);
+                break;
```
Should handling of the commands be the job of the parser? I feel it should be done in a separate class called handler or in the Boss class itself

SuhailLoya commented on src/main/java/boss/Ui.java:
```diff
@@ -0,0 +1,45 @@
+ * Represents the class that deals with the
+ * interactions with the user.
+ */
+public class Ui {
```
I like the neat and clean code :))

SuhailLoya commented on src/test/java/BossTest.java:
```diff
@@ -0,0 +1,66 @@
+     * Conducts a test to check whether the load function works.
+     */
+    @Test
+    public void testLoad() {
```
perhaps the test function name also describe the scenario beign tested and expected output

> featureUnderTest_testScenario_expectedBehavior()

SuhailLoya reviewed (commented):
Overall LGTM! Just a few nits to fix

leowolf275 committed:
Implement a GUI for chatbot

leowolf275 committed:
Merge branch 'branch-Level-10'

leowolf275 committed:
Add Reminder extension to chatbot

leowolf275 committed:
Merge tag 'B-Reminders'

leowolf275 committed:
Add assertions to program code

leowolf275 committed:
Improve code quality

leowolf275 committed:
Merge pull request #1 from leowolf275/branch-A-Assertions

Add assertions

leowolf275 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/boss/Boss.java
#	src/main/java/boss/Parser.java
#	src/main/java/boss/Storage.java
#	src/main/java/boss/TaskList.java

leowolf275 committed:
Merge pull request #2 from leowolf275/branch-A-CodeQuality

Improve code quality

leowolf275 committed:
Add personal chatbot responses

The current chatbot provides generic responses that lack a personal
touch.

A more personal experience will make the user more engaged.

Incorporate personal messages to deepen the connection with the user.

leowolf275 committed:
Move response functions to UI class for better OOP structure.

After changing the code for the GUI, the UI class is
not heavily utilized. The UI should be more response for
the chatbot responses.

Place these functions in the UI class will
improve the maintainability of the code.

Move relevant functions to the UI class to improve OOP.

leowolf275 committed:
Move more response functions to UI class

After changing the code for the GUI, the UI class is
not heavily utilized. The UI should be more response for
the chatbot responses.

Place these functions in the UI class will
improve the maintainability of the code.

Move relevant functions to the UI class to improve OOP.

leowolf275 committed:
Remove small mistakes in JavaDoc

leowolf275 committed:
Add more personal responses

leowolf275 committed:
Remove warnings in UI class

leowolf275 committed:
Add more JavaDoc

leowolf275 committed:
Improve remind feature

leowolf275 committed:
Update README.md

leowolf275 committed:
Add final adjustments

leowolf275 committed:
Fix bugs with regards to error messages

leowolf275 committed:
Update README.md

leowolf275 committed:
Add to Readme file
